core-ui/src/components/Predefined/Create/DNSEntries/DNSEntries.create.js

```diff
@@ -75,7 +75,7 @@ export function DNSEntriesCreate({
         id="dns-name-ref"
         domain={dnsEntry?.spec.dnsName}
         onChange={domain => {
-          jp.value(dnsEntry, '$.spec.dnsName', domain);
+          jp.value(dnsEntry, '$.spec.dnsName', domain.key);
           setDnsEntry({ ...dnsEntry });
         }}
```

Comment:
This bit should be handled by DNSNameRef internally - see relevant suggestions in that component.
```suggestion
```
Similar story with `TargetRef` and `TextRef`